### DISCUSSION TITLE
Auto Glossary

### DISCUSSION THREAD

**Discussion by AkiraVoid** [toeverything/AFFiNE#6179] [Ideas]
### Description

We need a glossary that records the terms and their meanings, which could be used to explain the keywords in a document.

我们需要一个记录术语以及它们的意义的术语表，该表可用于解释文档中的某些关键词。

### Use case

To say we have a document which is full of tech terms, and makes it be hard to understand for the people who are not familiar with tech. Even ones not familiar with some of the certain areas in tech may be confused. In such cases, we may want a glossary that, the ones who wrote this document could add terms and their meanings into it, then the one who read this document could see a brief meaning of a term by just putting the cursor onto that term.

想象我们有一篇全是科技术语的文档。这样的文档对于不熟悉科技领域的人来说理解起来十分困难，甚至对于那些只是没接触过部分领域的人来说也可能出现理解问题。这些情况下，我们会想要一个术语表，这样写文档的人能将文档中的术语及解释记录到术语表中，阅读文档的人也能通过将光标放置于其上来查看每个术语的意义。

### Anything else?

You may reference to the knowledge base of Lark (a.k.a Feishu).

你们可以参考飞书的知识库功能。

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by EYHN** [comment]
Whether the `link to docs` can satisfy the request of reference



![Image](https://github.com/toeverything/AFFiNE/assets/13579374/7ccb10aa-bd54-4fb8-8704-cba3a6d3dc26)

[a comment by an automated account is not shown]

**Comment by AkiraVoid** [comment]
> Whether the `link to docs` can satisfy the request of reference
> 
> 
> 
> ![Image](https://github.com/toeverything/AFFiNE/assets/13579374/7ccb10aa-bd54-4fb8-8704-cba3a6d3dc26)
> 
> 

I think it's hardly can. With glossary, the terms and explanations are not something that authors should do extra effort to make them visible to readers while writing the document, but a fully automatic process. You could imagine that some of the authors may not aware that there's a function like glossary, or they may know it exists, but don't know if there's such terms that they need. With glossary, every term detected in a document will be rendered with a pop-up that hosts its explanation, automatically. Besides all of this, we also want a centralized place to store and manage the terms, considering that there may be a same term that refers to completely different meanings in different areas. We also want to know who created this term, who added the explanation for the term and who can tell us more details about this term. With a document reference we may have some tricky workaround with these issues, but it is not designed for this. We don't want to let readers reference to a different document for more information, and the readers also don't know why they should open a different page for just a brief explanation.

**Comment by AkiraVoid** [comment]
I think I should be clearer about this idea.

**Features I need**
- Automatic highlighting of terms in documents, and show explanations while interacting with terms.
- One term with different explanations in different areas.
- One explanation with different terms (a term can have various forms, like JS and JavaScript).
- Search terms with global/union search.
- Add translations to a term.

**Comment by EYHN** [comment]
Thank you for your suggestion.

I think I have a clear idea of the feature you want. Due to the amount of work in this feature, I cannot give you a specific time for us to implement it.

For at least the next month our development and design resources will be focused on AI and improving the basic user experience.

I'll leave this issue here so we can look at it again in a month. 🫡

**Comment by Danie10** [comment]
When I used Joplin Notes I see it had something called Definition Lists. They formatted it like this:

```
First Term
: This is the definition of the first term.
Second Term
: This is one definition of the second term.
: This is another definition of the second term.
```
